pkg/components/bindings/registry.go

```diff
@@ -46,24 +46,42 @@ type (
 		CreateOutputBinding(name, version string) (bindings.OutputBinding, error)
 	}
 
+	stringOrSliceOfStrings interface {
+		string | []string
+	}
+
 	bindingsRegistry struct {
 		inputBindings  map[string]func() bindings.InputBinding
 		outputBindings map[string]func() bindings.OutputBinding
 	}
 )
 
 // NewInput creates a InputBinding.
-func NewInput(name string, factoryMethod func() bindings.InputBinding) InputBinding {
+func NewInput[T stringOrSliceOfStrings](name T, factoryMethod func() bindings.InputBinding) InputBinding {
```

Comment:
Disclaimer: I dislike generics in general, not just in Go.

While I wouldn't put a blank ban for generics in Dapr, this usage in particular is a bit of a smell in my opinion, where the developer reasons about the valid input to the method by looking at the name, in this case `stringOrSliceOfStrings`.

What's your take @artursouza? You hail from a Java background so your input is valuable here.

Reply:
Are you suggesting that the naming could be improved (such as `nameOrNames` for the argument, or changing the type `stringOrSliceOfStrings`)... Or that I shouldn't have used generics at all?

Option 2 I was considering was changing the method signature to `New(name string, factoryMethod func(), aliases ...string)`, where `aliases` was optional and could have been left empty. But I disliked that `name` and `aliases` were separated by `factoryMethod` (of a different type).

Reply:
`New(name string, factoryMethod func(), aliases ...string)` looks like a much better alternative to me.

It also removes the boilerplate code of the switch case statement and allows the method to focus on business logic instead.

Reply:
I can make that change. The run with generics was fun while it lasted :)

Reply:
:)

Reply:
Made the change.

Reply:
This allows backwards compatibility to allow this method to receive both an array and a string. Generics fits well here because Go does not allow method overloading.